src/main/java/com/squareup/javapoet/LineWrapper.java

```diff
@@ -49,7 +56,7 @@
   void append(String s) throws IOException {
     if (closed) throw new IllegalStateException("closed");
 
-    if (indentLevel != -1) {
+    if (nextFlush != null) {
```

Comment:
`nextFlush != null` was way more clear to me than `indentLevel != -1`. I can't tell you how long I spent staring at the `indentLevel != -1` looking confused at what it was doing.